src/components/tab-group/tab-group.ts

```diff
@@ -0,0 +1,120 @@
+import {Component, Input, ViewChildren, NgZone} from '@angular/core';
+import {QueryList} from '@angular/core';
+import {ContentChildren} from '@angular/core';
+import {PortalHostDirective} from '../../core/portal/portal-directives';
+import {MdTabLabel} from './tab-label';
+import {MdTabContent} from './tab-content';
+import {MdTabLabelWrapper} from './tab-label-wrapper';
+import {MdInkBar} from './ink-bar';
+
+/**
+ * Used to generate unique ID's for each tab component
+ * @type {number}
+ */
+let nextId = 0;
+
+@Component({
+  selector: 'md-tab-group',
+  templateUrl: './components/tab-group/tab-group.html',
+  styleUrls: ['./components/tab-group/tab-group.css'],
+  directives: [PortalHostDirective, MdTabLabelWrapper, MdInkBar],
+})
+export class MdTabGroup {
+  @ContentChildren(MdTabLabel) labels: QueryList<MdTabLabel>;
+  @ContentChildren(MdTabContent) contents: QueryList<MdTabContent>;
+  @ViewChildren(MdTabLabelWrapper) labelWrappers: QueryList<MdTabLabelWrapper>;
+  @ViewChildren(MdInkBar) inkBar: QueryList<MdInkBar>;
+
+  @Input() selectedIndex: number = 0;
+
+  private _focusIndex: number = 0;
+  private _groupId: number;
+
+  constructor(private _zone: NgZone) {
+    this._groupId = nextId++;
+  }
+
+  /**
+   * Waits one frame for the view to update, then upates the ink bar
+   * Note: This must be run outside of the zone or it will create an infinite change detection loop
+   */
+  ngAfterViewChecked(): void {
+    this._zone.runOutsideAngular(() => {
+      window.requestAnimationFrame(() => {
```

Comment:
It should work to just say `requestAnimationFrame` without explicitly dereferencing `window`.